SU-SWS/stanford_events_importer
# D8CORE-5170

# READY FOR REVIEW

# Summary
- Update URLs from `events.stanford.edu` to `events-legacy.stanford.edu`
    - Note: I believe that these are mirrors of each other, currently, so functionality should be the same

# Needed By (Date)
- 1.15.2022?

# Criticality
- How critical is this PR on a 1-10 scale? 4/10
- We want to get this in place before the Localist launch in February

# Steps to Test

1. Install a D7 site and enable `stanford_events_importer` (JSA profile should do that automatically)
2. Create a Stanford Events Importer node, importing _Unlisted_ events from the "Stanford Web Services" organization
3. Ensure that https://events.stanford.edu/events/938/93837/ gets imported
2. Check out this branch
3. Run `drush updb`
4. Make an edit to the "JCB Test event" on `events.stanford.edu`. (Let's see if this link works: https://events.stanford.edu/eventadmin/EditEvent.do?EventID=93837)
5. Wait until the next push from `events.stanford.edu` (8AM, Noon, 4PM, 8PM)
6. Run cron
7. Check that your changes on `events.stanford.edu` were propagated to your Drupal site
8. Create a new Stanford Events Importer node, with an arbitrary category or organization (I often choose the category "Film")
9. Ensure that events are imported
10. Review the code changes - does it make sense to update the URLs to `events-legacy.stanford.edu` even in the help text and README documentation?

## Additional Questions to Ponder

1. Do we make this a new major version? (7.x-4.x) See Shea's comments, below.
2. Do we care that we're changing variables that users may have set (`stanford_events_category_source` and `stanford_events_organization_source`)? (I think it's highly unlikely)

# Affected Projects or Products
- ACSF
- StanfordPro
- Other D7 sites that import from `events.stanford.edu`
- Localist

# Associated Issues and/or People
- JIRA ticket: `D8CORE-5170`
- Other PRs

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
